src/lib/Builder.hs

```diff
@@ -59,7 +59,8 @@ import PPrint
 import Util (bindM2, scanM, restructure)
 
 newtype BuilderT m a = BuilderT (ReaderT BuilderEnvR (CatT BuilderEnvC m) a)
-  deriving (Functor, Applicative, Monad, MonadIO, MonadFail, Alternative)
+  deriving ( Functor, Applicative, Monad, MonadIO, MonadFail, MonadWriter w
```

Comment:
What's `w` here?

Reply:
I believe it's implicitly a "forall `w`".